src/sensor_plugin.cpp

```diff
@@ -56,3 +56,13 @@ bool SensorPlugin::updated() {
   }
   return false;
 }
+
+bool SensorPlugin::updatedGPS() {
```

Comment:
Correct me if I am wrong, but this seems like a re-implementation of the `updated` function in the `sensor_plugin` base class. (https://github.com/PX4/px4-jsbsim-bridge/blob/d2cc31d7171155bdfec72ba73850288e40242f81/src/sensor_plugin.cpp#L50 ) Could you explain why re-implementing this is necessary? Could it be that it was just not being initialized properly in the base class?

Reply:
I believe what is occurring: that the _last_sim_time variable is either not being initialized correctly or? but could not find an alternate solution to what I proposed. 

**Here is the observation without proposed changes (and debug print messages):**

![Screenshot from 2020-09-29 06-12-23](https://user-images.githubusercontent.com/24363960/94545610-c1dfb180-021a-11eb-9f8a-acc5f1de935f.png)

What I observe:

- Update Imu is called: jsbsim_bridge
- sensor_plugin checks _last_sim_time 
- Update mag is called: jsbsim_bridge
- sensor_plugin checks _last_sim_time
-- because sensor_mag_plugin constructor has another declaration of _last_sim_time it always stays zero
- Update baro is called: jsbsim_bridge
- sensor_plugin checks _last_sim_time
- Update Gps is called: jsbsim_bridge
- sensor_plugin checks _last_sim_time

The GPS _last_sim_time is nan and therefore if _update_rate is anything other than 0 it will not publish GPS

The problem is bypassed for the mag sensor logic because the _update_rate is defaulted to 0.